allennlp/modules/backbones/vilbert_backbone.py

```diff
@@ -123,7 +121,32 @@ def forward(
         # Shape: (batch_size, num_tokens)
         attention_mask = text["tokens"].get("mask")
 
-        # Shape: (batch_size, num_tokens, embedding_dim)
+        dimensions = box_features.shape
+        batch_size = dimensions[0]
+        feature_size = dimensions[-1]
+        rolled_dimensions = dimensions[1:-2]
+        rolled_dimensions_product = 1
+        for dim in rolled_dimensions:
+            rolled_dimensions_product *= dim
+
+        if rolled_dimensions:
+            token_ids = token_ids.repeat_interleave(rolled_dimensions_product, 0)
```

Comment:
Why is this `repeat_interleave()`? Why can't it be done with `view()`?

Reply:
We're not actually repeating anything, are we?

Reply:
We actually are -- We need the text and image embeddings to have the same first dimension (batch_size * other dimensions), so I'm repeating the token ids however many times we need to so they match.

E.g. Our inputs are (16, 4, 100, 1024) and we roll the dimensions to get (64, 100, 1024), we'll still only have 16 sets of text tokens, so I repeat them 4 times. I guess I could do that directly on the embedding_output instead of the token_ids and token_type_ids, but this made logical sense to me as I was writing it.

Reply:
I don't get it. If we have a (64, 100, 1024) tensor, isn't that 64 sequences of length 100 each?

Reply:
Resolved offline, now we require the tokens and images have the same rolled dimensions